source/s2n/s2n_tls_channel_handler.c

```diff
@@ -1019,6 +1076,50 @@ static int s_s2n_handler_shutdown(
             s2n_handler->state = NEGOTIATION_FAILED;
         }
 
+        if (!abort_immediately && s2n_handler->state == NEGOTIATION_SUCCEEDED &&
+            !aws_linked_list_empty(&s2n_handler->input_queue) && slot->adj_right) {
+            /**
+             * In case of if we have any queued data in the handler after negotiation and we start to shutdown,
+             * make sure we pass those data down the pipeline before we complete the shutdown.
+             */
+            size_t downstream_size = aws_channel_slot_downstream_read_window(slot);
```

Comment:
trivial: I liked how you moved this to a little utility function in the darwin code, to keep the if/else's simpler

Reply:
yeah, I haven't made the change we agreed for drawin to s2n yet... should notify you earlier😣